modules/enterprise/server/jar/src/main/java/org/rhq/enterprise/server/rest/SetCallerInterceptor.java

```diff
@@ -72,6 +81,14 @@ public Object setCaller(InvocationContext ctx) throws Exception {
 
         Subject caller=null;
         java.security.Principal p = ejbContext.getCallerPrincipal();
+        if (!startupBean.isInitialized()) {
+            String notInitMessage = "Tried to call REST endpoint but the server has not finished to startup";
+            log.warn(notInitMessage);
```

Comment:
I changed the message to debug, Is not that this is not running yet or not running at all, but it's in the initialization process, may be loading some database configurations, but the REST API is already deployed and the EAP server is already running so the message is logged property (at least on my tests).

In this particular case, happens that `getDefaultDatabaseType` attribute is null because isn't set yet when someone tries to consume the REST service.